src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OpenIDAuthentication.java

```diff
@@ -0,0 +1,46 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc;
+
+import java.io.IOException;
+
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import fish.payara.security.annotations.LogoutDefinition;
+import fish.payara.security.annotations.OpenIdAuthenticationDefinition;
+import fish.payara.security.openid.api.OpenIdConstant;
+import jakarta.annotation.security.DeclareRoles;
+import jakarta.ejb.EJB;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.HttpConstraint;
+import jakarta.servlet.annotation.ServletSecurity;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+/**
+ * OIDC login implementation
+ */
+@WebServlet("/oidc/login")
+@OpenIdAuthenticationDefinition(
+    providerURI = "#{openIdConfigBean.providerURI}",
+    clientId = "#{openIdConfigBean.clientId}",
+    clientSecret = "#{openIdConfigBean.clientSecret}",
+    redirectURI = "#{openIdConfigBean.redirectURI}",
+    logout = @LogoutDefinition(redirectURI = "#{openIdConfigBean.logoutURI}"),
+    scope = {OpenIdConstant.OPENID_SCOPE, OpenIdConstant.EMAIL_SCOPE, OpenIdConstant.PROFILE_SCOPE},
```

Comment:
This should be configurable! I know IdPs out there that require adding custom scopes and it would be a real bummer if people are stuck because we hardcoded these.

Reply:
It was hard coded before, it can be made configurable at later point.